lib/spack/spack/spec.py

```diff
@@ -2665,6 +2665,7 @@ def virtual_dependencies(self):
         return [spec for spec in self.traverse() if spec.virtual]
 
     @property
+    @memoized
```

Comment:
@michaelkuhn Do you remember if there was an inherent reason to add memoization here?